examples/camera-app/esp32/main/Kconfig.projbuild

```diff
@@ -0,0 +1,129 @@
+#
+#    Copyright (c) 2021-2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+#    Description:
+#      Configuration options CHIP ESP32 demo application.
+#
+
+menu "Demo"
+
+    choice
+        prompt "Device Type"
+        default DEVICE_TYPE_ESP32_DEVKITC if IDF_TARGET_ESP32
+        default DEVICE_TYPE_ESP32_C3_DEVKITM if IDF_TARGET_ESP32C3
+        default DEVICE_TYPE_ESP32_S3_DEVKITM if IDF_TARGET_ESP32S3
+        default DEVICE_TYPE_ESP32_C2_DEVKITM if IDF_TARGET_ESP32C2
+        default DEVICE_TYPE_ESP32_C6_DEVKITC if IDF_TARGET_ESP32C6
+        default DEVICE_TYPE_ESP32_H2_DEVKITC if IDF_TARGET_ESP32H2
+        help
+            Specifies the type of ESP32 device.
+
+            Note that the "ESP32-DevKitC" choice is compatible with a number of clone devices
+            available from third-party manufacturers.
+
+        config DEVICE_TYPE_ESP32_DEVKITC
+            bool "ESP32-DevKitC"
+            depends on IDF_TARGET_ESP32
+        config DEVICE_TYPE_ESP32_WROVER_KIT
+            bool "ESP32-WROVER-KIT_V4.1"
+            depends on IDF_TARGET_ESP32
+        config DEVICE_TYPE_M5STACK
+            bool "M5Stack"
+            depends on IDF_TARGET_ESP32
+        config DEVICE_TYPE_ESP32_C3_DEVKITM
+            bool "ESP32C3-DevKitM"
+            depends on IDF_TARGET_ESP32C3
+        config DEVICE_TYPE_ESP32_S3_DEVKITM
+            bool "ESP32S3-DevKitM"
+            depends on IDF_TARGET_ESP32S3
+        config DEVICE_TYPE_ESP32H2_DEVKITC
+            bool "ESP32H2-DevKitC"
+            depends on IDF_TARGET_ESP32H2
+        config DEVICE_TYPE_ESP32_C2_DEVKITM
+            bool "ESP32C2-DevKitM"
+            depends on IDF_TARGET_ESP32C2
+        config DEVICE_TYPE_ESP32_C6_DEVKITC
+            bool "ESP32C6-DevKitC"
+            depends on IDF_TARGET_ESP32C6
+    endchoice
+
+    config CHIP_PROJECT_CONFIG
+        string "CHIP Project Configuration file"
+        default "main/include/CHIPProjectConfig.h"
+
+    choice
```

Comment:
even this and the `RENDEZVOUS_MODE` as we generally generate the onboarding code outside of the ðevice